dataprofiler/profilers/profile_builder.py

```diff
@@ -1355,11 +1359,125 @@ def _update_row_statistics(self, data, sample_ids=None):
             self.row_has_null_count = len(null_in_row_count)
             self.row_is_null_count = len(null_rows)
 
+    def _get_correlation(self, clean_samples):
+        """
+        Calculate correlation matrix on the cleaned data.
+
+        :param clean_samples: the input cleaned dataset
+        :type clean_samples: dict()
+        """
+        columns = self.options.correlation.columns
+        if columns is None:
+            for col in self._profile:
+                if (self._profile[col].profile['data_type'] not in ['int', 'float']
+                        or self._profile[col].profile['statistics']['null_count'] > 0):
+                    clean_samples.pop(col)
+        if len(clean_samples) <= 1:
+            return
+
+        data = pd.DataFrame(clean_samples)
+        data = data.apply(pd.to_numeric, errors='coerce')
+        self.correlation_matrix = data.corr()
+
+    def _update_correlation(self, clean_samples):
+        """
+        Update correlation matrix for cleaned data.
+
+        :param clean_samples: the input cleaned dataset
+        :type clean_samples: dict()
+        """
+        if self.correlation_matrix is not None:
```

Comment:
maybe it should be based on samples processed as oppose to `None` this would fix the other issue and allow it to be set to `None` still.

Reply:
imo None here means either we don't have any data yet, or the current data doesn't have enough numerical data to calculate the correlation. So, I don't really see issue with this.

Reply:
The problem becomes if you merge profiles and the value gets set as None bc they don't overlap in sets, if you then update the merged profile, it will then set the correlation because it is None.

Really, it should only update if the match count has not yet been updated since we are disabling updates after the first profile.

Reply:
changed